File: Array/Pt/ques3.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> store;

        for (int i=0; i<numRows; i++){
            vector<int> row;
            for (int  j=0; j<=i; j++){
                if (j==0|| j==i){
                    row.push_back(1);
                }
                else {
                    row.push_back(store[i-1][j-1] +store[i-1][j]);
                }
            }
            store.push_back(row);
        }
        return store;
    }
};

File: Array/Pt/ques3.js
/**
 * 
 * 118. Pascal's Triangle

 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {


    let res = []
  

    for(let i=0;i<numRows;i++){
        let row = [];
        for(let j=0;j<=i;j++){
            if(j==0 || j==i){
                row.push(1);
            }else{
                row.push(res[i-1][j-1]+res[i-1][j])
            }
        }
        res.push(row)
    }

    return res;
};